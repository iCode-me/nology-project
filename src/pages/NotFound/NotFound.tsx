import { Link } from "react-router-dom";
import Layout from "../../components/Layout/Layout";
import { ErrorCode } from "../../enums/ErrorCode";

const NotFoundView = () => (
  <Layout>
    <h1>Error {ErrorCode.NOT_FOUND} - Not Found!</h1>
    <Link to="/">Home</Link>
  </Layout>
);

export default NotFoundView;
