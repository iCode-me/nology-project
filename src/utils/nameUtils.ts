export const formatName = (firstName: string, lastName: string): string => {
  return `${firstName} ${lastName}`;
};

// export const formatNameArray = (names: string[]): string => {
//   return names.join(" ");
// };
