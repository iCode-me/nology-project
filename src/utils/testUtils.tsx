import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

type CustomRenderOptions = {
  useRouting: boolean;
};

export const customRender = (
  ui: JSX.Element,
  options: CustomRenderOptions = {
    useRouting: true,
  }
) => {
  let uiResult = ui;

  if (options.useRouting) {
    uiResult = wrapWithRouting(ui);
  }

  return render(uiResult);
};

const wrapWithRouting = (ui: JSX.Element): JSX.Element => {
  return <BrowserRouter>{ui}</BrowserRouter>;
};
