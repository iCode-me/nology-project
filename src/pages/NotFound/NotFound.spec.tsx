import { customRender } from "../../utils/testUtils";
import NotFound from "./NotFound";

export default describe("NotFoundView", () => {
  it("Renders", () => {
    const { container } = customRender(<NotFound />);
    expect(container).toMatchSnapshot();
  });
});
