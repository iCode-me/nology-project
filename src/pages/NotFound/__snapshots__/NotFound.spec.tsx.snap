// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundView Renders 1`] = `
<div>
  <div
    class="layout"
  >
    <main
      class="layout__main"
    >
      <h1>
        Error 
        404
         - Not Found!
      </h1>
      <a
        href="/"
      >
        Home
      </a>
    </main>
  </div>
</div>
`;
